slowly-changing-lookup-cache/README.md

```diff
@@ -0,0 +1,70 @@
+# slowly-changing-lookup-cache
+An implementation of "Pattern: Slowly-changing lookup cache" pattern of [Guide to common Cloud Dataflow use-case patterns](https://cloud.google.com/blog/big-data/2017/06/guide-to-common-cloud-dataflow-use-case-patterns-part-1).
+
+## What is this
+
+* Reading a streaming data from Cloud PubSub as `mainInput` for pipeline. 
+* Enriching `mainInput` data with `sideInput`.
+    - In this example, both inputs are `string`, we merging these two and showing as an log.
+* `sideInput` can be any external resources. For the convenience, reading a text file from Cloud Storage.
+* `sideInput` is a slow changing data. We are updating this every `N` seconds. 
+
+<img src="console.png" width="600">
+
+## How to Run
+
+0. Please substitute appropriate values
+    - `${BUCKET}`: Cloud Storage Bucket name.
+    - `${RUNNER}`: Beam Runner name. ex: DirectRunner, DataflowRunner
+    - `${GOOGLE_PROJECT}`: Google Project name.
+    - `${PUBSUB_TOPIC_NAME}`: Cloud Pubsub Topic name.
+    
+1. Prepare a text file, upload to Cloud Storage.
+```bash
+echo "world" > data_side_input.txt
+
+gsutil cp data_side_input.txt gs://${BUCKET}/
```

Comment:
Can we specify `projet_id` in gsutil command?

Reply:
Yes we could. 
```
gsutil --project ${PROJECT_ID} cp hogehoge gs://hogehoge
```